libraries/net/eth/EthernetInterface/EthernetInterface.cpp

```diff
@@ -92,12 +94,15 @@ int EthernetInterface::init() {
     return init_netif(NULL, NULL, NULL);
 }
 
-int EthernetInterface::init(const char* ip, const char* mask, const char* gateway) {
+int EthernetInterface::init(const char* ip, const char* mask, const char* gateway, const char * dns) {
```

Comment:
This API change appears unrelated to the issue being discussed.  IMHO it would be better as a separate PR.